jcl/src/java.base/share/classes/java/lang/System.java

```
@@ -397,41 +395,84 @@ private static void ensureProperties() {
 	StringBuffer.initFromSystemProperties(systemProperties);
 	StringBuilder.initFromSystemProperties(systemProperties);
-	/*[ENDIF]*/
+/*[ENDIF] Sidecar19-SE */
+/*[ENDIF] Java12 */
 
-	String javaRuntimeVersion = systemProperties.getProperty("java.runtime.version"); //$NON-NLS-1$
+/*[IF Java12]*/
+	String javaRuntimeVersion = initializedProperties.get("java.runtime.version"); //$NON-NLS-1$
+/*[ELSE]
+	String javaRuntimeVersion = initializedProperties.getProperty("java.runtime.version"); //$NON-NLS-1$
+/*[ENDIF]*/
 	if (null != javaRuntimeVersion) {
-		String fullVersion = systemProperties.getProperty("java.fullversion"); //$NON-NLS-1$
+	/*[IF Java12]*/
+		String fullVersion = initializedProperties.get("java.fullversion"); //$NON-NLS-1$
+	/*[ELSE]
+		String fullVersion = initializedProperties.getProperty("java.fullversion"); //$NON-NLS-1$
+	/*[ENDIF]*/
 		if (null != fullVersion) {
-			systemProperties.put("java.fullversion", (javaRuntimeVersion + "\n" + fullVersion)); //$NON-NLS-1$ //$NON-NLS-2$
+			initializedProperties.put("java.fullversion", (javaRuntimeVersion + "\n" + fullVersion)); //$NON-NLS-1$ //$NON-NLS-2$
 		}
 		rasInitializeVersion(javaRuntimeVersion);
 	}
 
-	lineSeparator = systemProperties.getProperty("line.separator", "\n"); //$NON-NLS-1$
+/*[IF Java12]*/
+	lineSeparator = initializedProperties.getOrDefault("line.separator", "\n"); //$NON-NLS-1$
+/*[ELSE]
+	lineSeparator = initializedProperties.getProperty("line.separator", "\n"); //$NON-NLS-1$
+/*[ENDIF]*/
 	/*[IF Sidecar19-SE]*/
-	useLegacyVerPresents = systemProperties.containsKey("use.legacy.version");
+	useLegacyVerPresents = initializedProperties.containsKey("use.legacy.version");
 	/*[ENDIF]*/
+
+	if (isInitialization) {
+		/*[PR CMVC 179976] System.setProperties(null) throws IllegalStateException */
+		try {
+		/*[IF Java12]*/
+			VM.saveProperties(initializedProperties);
+		/*[ELSE]
+			VM.saveAndRemoveProperties(initializedProperties);
+		/*[ENDIF]*/
+		} catch (NoSuchMethodError e) {
```

Please add a comment explaining why this exception is being ignored.

---

I actually have no idea what this is for. This was something I copied over when I refactored this method to make it compatible with 12. I can see what happens when its missing in an 8 build..

---

I've applied your suggested changes in a new commit except for this one atm. thanks for the review

---

I suggest we remove the try-catch surroundings: It's hard to imagine why `NoSuchMethodError` would be thrown and, for me, harder to explain why it should be ignored.

---

okay. it has been removed.